internal/cmd/update/command.go

```diff
@@ -89,89 +65,197 @@ func (c *command) update(cmd *cobra.Command, _ []string) error {
-	if !c.client.PromptToDownload(pversion.CLIName, c.version.Version, "v"+updateVersion, releaseNotes, !yes) {
-		return nil
+	if !yes {
+		ok, err := verify()
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return nil
+		}
 	}
 
-	oldBin, err := os.Executable()
+	assets, _, err := client.Repositories.ListReleaseAssets(context.Background(), pgithub.Owner, pgithub.Repo, releases[len(releases)-1].GetID(), nil)
 	if err != nil {
 		return err
 	}
-	if err := c.client.UpdateBinary(pversion.CLIName, updateVersion, oldBin, noVerify); err != nil {
-		return errors.NewUpdateClientWrapError(err, errors.UpdateBinaryErrorMsg)
+
+	var binaryReader io.Reader
+	var bytes int64
+	var checksum []byte
+
+	binary := fmt.Sprintf("confluent_%s_%s_%s", latestVersion.String(), runtime.GOOS, runtime.GOARCH)
+
+	for _, asset := range assets {
+		if asset.GetName() == binary {
+			binaryReader, bytes, err = downloadReleaseAsset(client, asset)
+			if err != nil {
+				return err
+			}
+		} else if !noVerify && asset.GetName() == fmt.Sprintf("confluent_%s_checksums.txt", latestVersion.String()) {
+			reader, _, err := downloadReleaseAsset(client, asset)
+			if err != nil {
+				return err
+			}
+
+			checksums, err := io.ReadAll(reader)
+			if err != nil {
+				return err
+			}
+
+			checksum, err = findChecksum(string(checksums), binary)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if binaryReader == nil {
+		return fmt.Errorf("could not find release")
+	}
+	if !noVerify && checksum == nil {
+		return fmt.Errorf("could not find checksum")
 	}
 
+	var opts update.Options
+	if !noVerify {
+		opts.Checksum = checksum
+	}
+
+	start := time.Now()
+	if err := update.Apply(binaryReader, opts); err != nil {
+		return err
+	}
+
+	mb := convertBytesToMegabytes(bytes)
+	s := time.Since(start).Seconds()
+	output.Printf("Done. Downloaded %.2f MB in %.0f seconds. (%.2f MB/s)\n", mb, s, mb/s)
+
 	return nil
 }
 
-func (c *command) getReleaseNotes(cliName, latestBinaryVersion string) string {
-	latestReleaseNotesVersion, allReleaseNotes, err := c.client.GetLatestReleaseNotes(cliName, c.version.Version)
+// getRelevantReleases gets an ordered list of releases that have happened since the last update.
+func getRelevantReleases(releases []*github.RepositoryRelease, current semver.Version, major bool) []*github.RepositoryRelease {
+	var lo int
+	var hi int
 
-	var errMsg string
-	if err != nil {
-		errMsg = fmt.Sprintf(errors.ObtainingReleaseNotesErrorMsg, err)
-	} else {
-		isSameVersion, err := sameVersionCheck(latestBinaryVersion, latestReleaseNotesVersion)
+	var found bool
+
+	for i, release := range releases {
+		version, err := getReleaseVersion(release)
 		if err != nil {
-			errMsg = fmt.Sprintf(errors.ReleaseNotesVersionCheckErrorMsg, err)
+			continue
 		}
-		if !isSameVersion {
-			errMsg = fmt.Sprintf(errors.ReleaseNotesVersionMismatchErrorMsg, latestBinaryVersion, latestReleaseNotesVersion)
+
+		if !found && (major || version.Major == current.Major) {
```

Comment:
Is `releases` initially newest-first? If so, I'm wondering what would happen if the user didn't choose `--major` and the 30th release satisfies `version.Major == current.Major`; it seems like we would get `lo == hi` and end up returning nothing.

If it's initially sorted oldest-first then ignore this comment.

Reply:
It's newest first.

> I'm wondering what would happen if the user didn't choose --major and the 30th release satisfies version.Major == current.Major

Yeah, we'd have to iterate through all 30 in that case. Not sure if there's any way we can only fetch minor versions... let me look into it, but I don't think it's possible.